### PR TITLE
Introduce f3d::window and improve libf3d API

### PR DESCRIPTION
**Pull request by mwestphal**
- Introduce f3d::window, f3d::windowStandard and f3d::windowNoRender
- Improve f3d::interactor a lot by moving many methods to it
- Add a Timer API to f3d::interactor
- Move F3DAnimationManager to f3d::interactor
- Remove many methods and logic from f3d::loader
- Recorder small cleanup

### CODE REVIEW

**Comment by mwestphal** [comment]
@Meakk : this defines f3d::window, f3d::windowNoRender and f3d::windowStandard. Not sure this is perfect but better than nothing.

**Comment by mwestphal** [comment]
@Meakk pls review

**Comment by mwestphal** [comment]
@Meakk ready for review

File: src/library/f3d_windowStandard.h [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+class vtkRenderWindow;
+class vtkF3DGenericImporter;
+namespace f3d
+{
+class loader;
+class windowStandard : public window
+{
+public:
+  windowStandard(const std::string& windowName, bool offscreen);
+  ~windowStandard();
```
Comment (Meakk) [inline src/library/f3d_windowStandard.h:18]: override

File: src/library/f3d_windowNoRender.h [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+class vtkRenderWindow;
+class vtkF3DRenderer;
+namespace f3d
+{
+class loader;
+class windowNoRender : public window
+{
+public:
+  windowNoRender();
+  ~windowNoRender();
```
Comment (Meakk) [inline src/library/f3d_windowNoRender.h:18]: override

**Review by Meakk — APPROVED** [review]
LGTM

**Pull request merged**